tests/unit/amo/components/TestSearchResult.js

```diff
@@ -131,19 +131,197 @@ describe(__filename, () => {
   });
 
   it('adds a theme-specific class', () => {
-    const addon = createInternalAddon({
-      ...fakeAddon,
-      type: ADDON_TYPE_THEME,
-      theme_data: {
-        previewURL:
-          'https://addons.cdn.mozilla.net/user-media/addons/334902/preview_large.jpg?1313374873',
-      },
+    const root = render({
+      addon: createInternalAddon({
+        ...fakeAddon,
+        type: ADDON_TYPE_THEME,
+        theme_data: {
+          previewURL:
+            'https://addons.cdn.mozilla.net/user-media/addons/334902/preview_large.jpg?1313374873',
+        },
+      }),
     });
-    const root = render({ addon });
 
     expect(root).toHaveClassName('SearchResult--theme');
   });
 
+  it('does not render an image if the isAllowedOrigin is false', () => {
+    const root = render({
+      addon: createInternalAddon({
+        ...fakeAddon,
+        type: ADDON_TYPE_STATIC_THEME,
+        previews: [
+          {
+            ...fakePreview,
+            thumbnail_url: 'http://example.com',
```

Comment:
How do we know this is not an allowed origin? We shouldn't write a test where we assume the setup will work. We need to be able to inject a value for `isAllowedOrigin` into the component in order to test this properly, and then use a `sinon.stub` in the test.

Reply:
hmm.. ok let me look into this more

Reply:
okay, I've updated this...let me know what you think : )